datafusion/physical-expr/src/expressions/binary.rs

```diff
@@ -2481,6 +2492,34 @@ mod tests {
         Ok(())
     }
 
+    #[test]
+    fn bitwise_shift_array_test() -> Result<()> {
+        let input = Arc::new(Int32Array::from(vec![Some(2), None, Some(10)])) as ArrayRef;
+        let modules =
+            Arc::new(Int32Array::from(vec![Some(2), Some(4), Some(8)])) as ArrayRef;
+        let mut result = bitwise_shift_left(input.clone(), modules.clone())?;
+
+        let expected = Int32Array::from(vec![Some(8), None, Some(2560)]);
+        assert_eq!(result.as_ref(), &expected);
+
+        result = bitwise_shift_right(result.clone(), modules.clone())?;
+        assert_eq!(result.as_ref(), &input);
+
+        Ok(())
+    }
+
+    #[test]
+    fn bitwise_shift_array_overflow_test() -> Result<()> {
+        let input = Arc::new(Int32Array::from(vec![Some(2)])) as ArrayRef;
+        let modules = Arc::new(Int32Array::from(vec![Some(100)])) as ArrayRef;
+        let result = bitwise_shift_left(input.clone(), modules.clone())?;
+
+        let expected = Int32Array::from(vec![Some(32)]);
```

Comment:
TIL `2 << 100` = `32` 🤯 

https://play.rust-lang.org/?version=stable&mode=release&edition=2018&gist=b0682ca51f798793e784b852e28dc4a8

Reply:
2<<100 =>> 2<<(100%bit_width(i32)) =>> 2 << 4?

Reply:
i am surprised that this is rotational shifting

Reply:
> i am surprised that this is rotational shifting

me too...